opm/simulators/utils/readDeck.cpp

```diff
@@ -402,15 +403,28 @@ void Opm::ensureOutputDirExists(const std::string& cmdline_output_dir)
     }
 }
 
+void Opm::prepareResultOutputDirectory(const std::string&           baseName,
+                                       const std::filesystem::path& outputDir)  
+{   
+    //Loop over all files in the output directory and subdirectories and delete them if their name is baseName + a correct extension
+    std::regex r(baseName + ".(F?(DBG|E?GRID|INIT|PRT|RFT|SMSPEC|UNSMRY|UNRST)|([ABCFGHSTUXYZ]\\d{4})|(INFOSTEP|INFOITER|OPMRST))");
+    for (auto& file : std::filesystem::recursive_directory_iterator(outputDir)) {
+        std::string fileName = file.path().filename();
+        if (std::regex_match(fileName, r)) {
+            std::filesystem::remove(file);
+        }
+    }
```

Comment:
One **slight** alternative could be to match the `baseName` by string equality instead of regular expressions, e.g., along the lines of
```C++
const std::regex extension(R"~(\.F?(DBG|E?GRID|INIT|....|INFOITER|OPMRST))~");
for (const auto& elem : std::filesystem::recursive_directory_iterator(outputDir)) {
    const auto& pth = elem.path();
    if ((pth.stem() == baseName) && std::regex_match(pth.extension(), extension)) {
        std::filesystem::remove(elem);
    }
}
```
but it's not clear to me that that would be better.  I would advise using a raw string here, however, since that would simplify entering metacharacters&ndash;i.e., not having to double every backslash character.  We might additionally want to check `elem.is_regular_file()` for some additional safety, although it's not clear to me what to do if we encounter a directory with a matching name.